packages/ui/src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,90 @@
+import { ComponentProps } from 'react';
+
+import { useTextFieldValidation } from '../../hooks/useTextFieldValidation';
+import { Icon } from '../Icon';
+import { Center, Flex } from '../Layout';
+import { Text } from '../Text';
+
+type Props = {
+  /**
+   * Input value to display in the input field component.
+   */
+  value: string;
+  /**
+   * Function to call when the reset button is clicked.
+   */
+  onClickReset: () => void;
+  /**
+   * Placeholder text to display in the input field component.
+   * @default ''
+   */
+  placeholder?: string;
+  /**
+   * Error message to display when validation fails.
+   * @default ''
+   */
+  errorMessage?: string;
+  /**
+   * Minimum length of the input value.
+   * @default 1
+   */
+  minLength?: number;
+  /**
+   * Maximum length of the input value.
+   * @default 10
+   */
+  maxLength?: number;
+} & ComponentProps<'input'>;
+
+export function TextField({
+  value,
+  onClickReset,
+  placeholder = '',
+  errorMessage = '',
+  minLength = 1,
+  maxLength = 10,
+  ...props
+}: Props) {
+  const { hasError, handleBlur } = useTextFieldValidation({
+    value,
+    minLength,
+  });
+
+  return (
+    <Flex className="flex-col items-start">
+      <Center className="relative">
+        <input
+          value={value}
+          placeholder={placeholder}
+          maxLength={maxLength}
+          className="h-11 w-[272px] rounded-lg py-[9px] pl-3 pr-10 text-gray-900 outline-none outline outline-1 outline-gray-200 transition-colors placeholder:text-gray-300"
+          onBlur={handleBlur}
```

Comment:
요거는 추후 onBlur props에 따라 덮여씌워질수 있을 것 같아요. 이런 식으로 한번 감싸주는게 안전할 것 같네요

```suggestion

          //props 입력 부분에서
          { value,
             onClickReset,
             placeholder = '',
             errorMessage = '',
             minLength = 1,
             maxLength = 10,
             onBlur, // onBlur 추가
             ...props
             }
          //...
          onBlur={(event) => {
            handleBlur(event);
            onBlur?.(event);
          }}
```

Reply:
[`6b22572`](https://github.com/dnd-side-project/dnd-12th-9-frontend/pull/27/commits/6b22572500cb43a8cdec82c037897e19b222b670) 수정했습니닷